compositor_render/src/sync_renderer.rs

```diff
@@ -29,7 +29,11 @@ impl SyncRenderer {
         let mut guard = self.0.lock().unwrap();
         match spec {
             RendererSpec::Shader(spec) => {
-                let shader = Arc::new(Shader::new(&ctx.wgpu_ctx, spec.source)?);
+                let shader = Arc::new(Shader::new(
+                    &ctx.wgpu_ctx,
+                    spec.source,
```

Comment:
maybe pass entire spec object instead?

Reply:
Can't do that, since builtins use this as well. I don't want to pass `ShaderSpec` to create Shader. `ShaderSpec` specifies `ShaderNode` params, while `Shader` is just abstraction over a single shader graphical pipeline.